articles/understanding-map-reduce-in-hadoop/index.md

```diff
@@ -0,0 +1,118 @@
+
+The following diagram shows an example of a MapReduce with the three main phases. Splitting is often included in the mapping stage. 
+
+![Example of MapReduce](/engineering-education/understanding-mapreduce-in-hadoop/example-of-mapreduce.png)
+
+[Image Source: Edureka](https://4zy7s42hws72i51dv3513vnm-wpengine.netdna-ssl.com/wp-content/uploads/2018/02/MapReduce-Way-MapReduce-Tutorial-Edureka.png)
+
+##### Combiner Phase
+This is an optional phase that is used for optimizing the MapReduce process. It is used for reducing the pap outputs at the node level. In this phase, duplicate outputs from the map outputs can be combined into a single output. The combiner phase increases speed in the Shuffling phase by improving the performance of Jobs. 
+
+The following diagram shows how all the four phases of MapReduce have been applied. 
+
+![MapReduce with Combiner Phase](/engineering-education/understanding-mapreduce-in-hadoop/mapreduce-with-combiner-phase.jpg)
+
+[Image Source: Cloud Front](https://d2h0cx97tjks2p.cloudfront.net/blogs/wp-content/uploads/sites/2/2017/05/mapreduce-job-execution-flow-1-1024x492-1.jpg)
+
+### Benefits of Hadoop MapReduce
+* **Speed:** MapReduce can process huge unstructured data within a short time. 
+* **Fault-tolerance:** The MapReduce framework can handle failures. 
+* **Cost-effective:** Hadoop has a scale-out feature that enables users to process or store data in a cost-effective manner. 
+* **Scalability:** Hadoop provides a highly scalable framework. MapReduce allows users to run applications from many nodes.
+* **Data availability:** Replicas of data are sent to various nodes within the network. This ensures copies of data are available even in the event of failure. 
+* **Parallel Processing:** In MapReduce, multiple job-parts of the same dataset can be processed in a parallel manner. This reduces the time taken to complete a task. 
+
+### Applications of Hadoop MapReduce
+The following are some of the practical applications of the MapReduce program.
+
+#### E-Commerce
+E-commerce companies such as Walmart, E-Bay, and Amazon use MapReduce to analyze buying behavior. MapReduce provides meaningful information that is used as the basis for developing product recommendations. Some of the information used include site records, e-commerce catalogs, purchase history, and interaction logs. 
+
+#### Social Networks
+The MapReduce programming tool can evaluate certain information on social media platforms such as Facebook, Twitter, and LinkedIn. It can evaluate important information such as who liked your status and who saw your profile. 
+
+#### Entertainment 
+MapReduce is used by Netflix to analyze the clicks and logs of online customers. This information helps the company to suggest movies based on customers’ interests and behavior. 

```

Comment:
Re-write in active voice: Netflix uses MapReduce to analyze the clicks and logs of online customers.

Reply:
Done.